source/Lib/vvdec/vvdec.cpp

```diff
@@ -377,16 +377,37 @@ vvdecNalType vvdec_get_nal_unit_type( vvdecAccessUnit *accessUnit )
   return vvdec::VVDecImpl::getNalUnitType(*accessUnit);
 }
 
-const char* vvdec_get_nal_unit_type_name( vvdecNalType t )
+VVDEC_DECL const char* vvdec_get_nal_unit_type_name( vvdecNalType t )
 {
   return vvdec::VVDecImpl::getNalUnitTypeAsString(t);
 }
 
-bool vvdec_is_nal_unit_slice( vvdecNalType t )
+VVDEC_DECL bool vvdec_is_nal_unit_slice( vvdecNalType t )
 {
   return vvdec::VVDecImpl::isNalUnitSlice(t);
 }
 
+VVDEC_DECL int vvdec_set_tracing( const char *file, const char *rule, const bool bPrint )
```

Comment:
Please use the following logic:

* if ENABLE_TRACING is used:
  * if file or rule not empty, initialize vvdec::g_trace_ctx and print out the channel list
  * else return VVDEC_OK straigt away
* else
  * if rule and file are empty, return VVDEC_OK
  * else return VVDEC_ERR_INITIALIZE

Reply:
* if ENABLE_TRACING is used
  - no matter 'file' or 'rule' is empty or not, vvdec::g_trace_ctx must be initialized, those MACROs would fail
   e.g. #define DTRACE(ctx,channel,...)              ctx->dtrace<true>( channel, __VA_ARGS__ )  
    (because ctx is nullptr)

Reply:
right.